### ISSUE TITLE
About Noise Settings

### ISSUE THREAD

**Issue by anomous123** [qiaoruiyt/NoiseRobustDG#1]
Dear authors,

Thanks for your open-source library. I would like to ask if the noise settings in the current code ensure that the noise labels remain consistent across different random seed selections. If not, would this affect the fairness of experimental comparisons?

Best regards.

**Comment by qiaoruiyt** [comment]
Hi! Thank you and that's a great question! Currently, the randomness of the noise depends on the random seed passed for training, so they can vary if the training seed changes. There are two ways of conducting the experiments: 
- Ideally, for each seed $i$, we would like to first sample a noisy dataset $\tilde{D}_i$, run the sweep and collect the performances for this seed, and finally aggregate the results across all seeds. This approach ensures better consistency and rigor, but it might too be costly to run. 
- Alternatively, as the scope is not to develop a new state-of-the-art algorithm, we can take a shortcut and randomly sample noise and hyperparameters at the same time to obtain some reasonable statistical estimates of the performance with less compute. 

Does that make sense?

**Comment by anomous123** [comment]
Thank you very much for your response! So, the results in your paper primarily follow Scheme 2, correct? If I want to conduct some explorations, which scheme do you think would be more reasonable?

**Comment by qiaoruiyt** [comment]
Yes, the paper adopts the second scheme. I would recommend trying out the first scheme if there is enough compute, though my guess is that the result won't be significantly different (but I might be wrong here). Looking forward to seeing the new results!

**Comment by anomous123** [comment]
Thank you very much for your suggestions.
